ds3-sdk/src/main/java/com/spectralogic/ds3client/metadata/MetaDataAccessImpl.java

```diff
@@ -0,0 +1,156 @@
+package com.spectralogic.ds3client.metadata;
+
+import com.spectralogic.ds3client.helpers.Ds3ClientHelpers;
+import org.apache.commons.io.FilenameUtils;
+
+import java.io.IOException;
+import java.nio.file.FileSystem;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.attribute.*;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Implementation of MetaDataAccess Interface
+ * Used to store meta data on Server
+ */
+public class MetaDataAccessImpl implements Ds3ClientHelpers.MetadataAccess {
+    private Map<String, Path> fileMapper = null;
+
+    public MetaDataAccessImpl(final Map<String, Path> fileMapper) {
+        this.fileMapper = fileMapper;
+    }
+
+    @Override
+    public Map<String, String> getMetadataValue(final String filename) {
+        Map<String, String> metadata = new HashMap<>();
```

Comment:
Why is this field initialized and then immediately reassigned on line 32?  This seems like an unnecessary object creation.  Also, exceptions should not be handled in this way.  At a minimum it should be logged with a slf4j logger, or wrapped in a runtime exception and re-thrown.  This code could be simplified to:
```java

try {
    final Path file = fileMapper.get(filename);
    return storeMetaData(file);
} catch (final Exception e) {
    // either log it
    LOG.error("Failed to store metadata", e);

    // or throw in a runtime exception
    throw new RuntimeException(e);
}

```